### ISSUE TITLE
[0.9.3] can not group by tag with period in the name

### ISSUE THREAD

**Issue by airyland** [influxdb/influxdb#4018]
I used page.group as tag key, but couldn't group by "page.group".

Server returned:

`select sum(page.value) from page group by "page.group"`

````
{"results":[{"series":[{"name":"page","tags":{"page.\"group\"":""},"columns":["time","sum"],"values":[["1970-01-01T00:00:00Z",1696]]}]}]}
````

![image](https://cloud.githubusercontent.com/assets/559179/9699084/6e43b476-5406-11e5-86a1-48df88c66893.png)

**Comment by beckettsean** [comment]
@airyland is `page.group` a field or a tag? GROUP BY currently only supports tags.

**Comment by beckettsean** [comment]
Also you need to double-quote `page.value` in the SELECT clause.

**Comment by airyland** [comment]
`page.group` is a tag.

select * from page

time	                                                    page.group	page.value
2015-09-08T21:03:32.725988377Z	   "hello"	        1
2015-09-08T21:04:09.66892902Z	   "hello2"	        1

select sum("page.value") from page group by "page.group"

{"results":[{"series":[{"name":"page","tags":{"page.\"group\"":""},"columns":["time","sum"],"values":[["1970-01-01T00:00:00Z",2]]}]}]}

SHOW TAG KEYS FROM page

{"results":[{"series":[{"name":"page","columns":["tagKey"],"values":[["page.group"]]}]}]}

**Comment by airyland** [comment]
select distinct(event.value) from event => [1099511758848, 1]
select * from event where event.value=1 => works
select * from event where event.value=1099511758848 => panic

**Comment by beckettsean** [comment]
@airyland is your most recent comment intended for https://github.com/influxdb/influxdb/issues/4020? I don't see how it is relevant to this one.

**Comment by beckettsean** [comment]
The problem is caused by the period in the tag name. Without the period everything works as intended.

Repro steps:

```
> insert page,page.group=hello page.value=1
> insert page,page.group=hello2 page.value=1
> select * from page
name: page
----------
time				page.group	page.value
2015-09-08T22:43:01.622617072Z	hello		1
2015-09-08T22:43:05.839069738Z	hello2		1

> select sum("page.value") from page group by "page.group"
name: page
tags: page."group"=
time			sum
----			---
1970-01-01T00:00:00Z	2
```

That last query should have returned two buckets for `page.group`, but instead it's using `page."group"`, which is a tag that doesn't exist. My best guess is the retention policy parsing is interfering, despite the tag key being double-quoted.

Here's the same data using a tag key without periods, and it works as expected (including the grouping where the tag is not set, which is what we see with the tag that has a period.)


```
> insert page,pagegroup=hello2 page.value=1
> insert page,pagegroup=hello1 page.value=1
> select sum("page.value") from page group by "pagegroup"
name: page
tags: pagegroup=
time			sum
----			---
1970-01-01T00:00:00Z	2


name: page
tags: pagegroup=hello1
time			sum
----			---
1970-01-01T00:00:00Z	1


name: page
tags: pagegroup=hello2
time			sum
----			---
1970-01-01T00:00:00Z	1

> 
```

**Comment by airyland** [comment]
@beckettsean Sorry, I made a mistake, it was the comment intended for #4020.

**Comment by corylanou** [comment]
This appears to be fixed now:

```
> insert page,page.group=hello page.value=1
> insert page,page.group=hello2 page.value=1
> select * from page
name: page
----------
time                            page.group      page.value
2015-11-17T18:14:46.497801849Z  hello           1
2015-11-17T18:14:52.873958535Z  hello2          1

> select sum("page.value") from page group by "page.group"
name: page
tags: page.group=hello
time                    sum
----                    ---
1970-01-01T00:00:00Z    1


name: page
tags: page.group=hello2
time                    sum
----                    ---
1970-01-01T00:00:00Z    1
```

Also confirmed admin shows proper results.
![screenshot 2015-11-17 12 17 32](https://cloud.githubusercontent.com/assets/585100/11220356/3f3958a4-8d25-11e5-9019-ed9381feee45.png)

@beckettsean can you confirm my results?  I want to make sure I understood the problem correctly.

**Comment by beckettsean** [comment]
@corylanou confirmed fixed in 0.9.4.2
